httpcore5/src/main/java/org/apache/hc/core5/reactor/SingleCoreIOReactor.java

```diff
@@ -315,6 +326,18 @@ private void validateAddress(final SocketAddress address) throws UnknownHostExce
     private void processPendingConnectionRequests() {
         IOSessionRequest sessionRequest;
         for (int i = 0; i < MAX_CHANNEL_REQUESTS && (sessionRequest = this.requestQueue.poll()) != null; i++) {
+
+            // Calculate wait time safely without keeping long-lived state
+            final long waitTimeMillis = System.currentTimeMillis() - sessionRequest.getEnqueueTime();
```

Comment:
@arturobernalg Would you agree to move metrics collection inside `if (threadPoolListener != null) {`? There is no need to collect them if there is no listener.

Reply:
@ok2c done